public/js/script.js

```diff
@@ -1,35 +1,34 @@
 var socket = io();
 $(document).ready(function() {
-  
+
   var id = "";
-  socket.emit('join');
   var mSecondsLeft = 0;
-  
+
+  socket.emit('join');
+
   socket.on('join', function(data) {
     id = data.id;
-    for (key in data.answers) {
-      for (key2 in data.answers[key]) {
+    for (var key in data.answers) {
```

Comment:
It's important to prefix your iterators with `var` like this. Otherwise you're making global vars - not a great idea.